org-formation/090-systems-manager/QuickSetup-PatchPolicy.yaml

```diff
@@ -0,0 +1,92 @@
+AWSTemplateFormatVersion: '2010-09-09'
+Description: An SSM patch policy to patch all instances in the given OUs
+Parameters:
+  TargetOrgRoot:
+    Type: String
+    Description: >-
+      AWS Organization root ID to target.
+  TargetRegions:
+    Type: String
+    Description: >-
+      Comma separated list of AWS regions to target.
+Resources:
+  PatchLogBucket:
+    Type: "AWS::S3::Bucket"
+    Properties:
+      PublicAccessBlockConfiguration:
+        BlockPublicAcls: true
+        BlockPublicPolicy: true
+        IgnorePublicAcls: true
+        RestrictPublicBuckets: true
+  PatchLogBucketPolicy:
+    Type: 'AWS::S3::BucketPolicy'
+    Properties:
+      Bucket: !Ref PatchLogBucket
+      PolicyDocument:
+        Version: '2012-10-17'
+        Statement:
+          # All member accounts need write access to this bucket
+          - Effect: Allow
+            Principal: '*'
+            Action:
+              - 's3:Get*'
+              - 's3:List*'
+              - 's3:Put*'
+            Resource:
+              - !Sub '${PatchLogBucket.Arn}'
+              - !Sub '${PatchLogBucket.Arn}/*'
+            Condition:
+              # require an account principal associated with our Org
+              StringEquals:
+                "aws:PrincipalOrgID": !Ref TargetOrgRoot
+              ArnLike:
+                "aws:PrincipalArn": "arn:aws:iam::*:root"
+  PatchPolicy:
+    Type: "AWS::SSMQuickSetup::ConfigurationManager"
+    Properties:
+      Name: "PatchResourceGroup"
+      Description: "Patch all SSM managed nodes"
+      ConfigurationDefinitions:
+      - Type: AWSQuickSetupType-PatchPolicy
+        LocalDeploymentExecutionRoleName: AWS-QuickSetup-StackSet-Local-ExecutionRole
+        LocalDeploymentAdministrationRoleArn: "arn:aws:iam::531805629419:role/AWS-QuickSetup-PatchPolicy-LocalAdministrationRole"
+        Parameters:
+          PatchPolicyName: PatchPolicyAll
+          # Patch baselines can be found with the command:
+          #   aws ssm describe-patch-baselines |jq '.BaselineIdentities | map({ (.OperatingSystem): (.value = .BaselineId | .label = .BaselineName | .description = .BaselineDescription | .disabled=false | del(.BaselineId, .BaselineName, .OperatingSystem, .BaselineDescription, .DefaultBaseline)) }) | add'
+          #
+          # More information at:
+          #   https://aws.amazon.com/blogs/mt/deploy-aws-systems-manager-quick-setup-programmatically-across-your-aws-organization/
+          #   https://github.com/aws-samples/aws-management-and-governance-samples/blob/master/AWSSystemsManager/Quick-Setup-API/patch-policy-examples/patch-policy-cfn-template.yaml
+          #   https://github.com/aws-samples/aws-management-and-governance-samples/blob/master/AWSSystemsManager/Quick-Setup-API/patch-policy-examples/default-patch-baselines.txt
+          ConfigurationOptionsPatchOperation: "ScanAndInstall"
```

Comment:
is this going to installing security patches only? or will it do more than that? Also do these patches require re-booting resources?

Reply:
Great question. The short answer is yes, it only installs security patches, but how they're selected depends on the flavor of Linux[1]. And there's a `RebootOption` parameter to enable rebooting when required[2]. I left it out to use the default of no reboot, but I can make that explicit.

[1] https://docs.aws.amazon.com/systems-manager/latest/userguide/patch-manager-selecting-patches.html
[2] https://docs.aws.amazon.com/quick-setup/latest/APIReference/API_ConfigurationDefinitionInput.html